autox/environment/nncf/quantization_env.py

```diff
@@ -0,0 +1,586 @@
+        self.qctrl            = quantization_controller
+        self.qmodel           = quantization_controller._model
+        self.criterion        = criterion
+        self.train_loader     = train_loader
+        self.val_loader       = val_loader
+        self.train_epoch_fn   = train_epoch_fn
+        self.validate_fn      = validate_fn
+
+        self.config = config
+        if self.config.get('nncf_config', None) is None:
+            raise KeyError("config must have nncf_config dictionary")
+        
+        # TODO: Do we need to a way to specify a different device for automated search?
+        # self.config.current_gpu = self.config.gpu_id
+        # torch.cuda.set_device(config.gpu_id)         # Set default operating cuda device
+        # self.config.device = get_device(self.config) # get_device requires config.current_gpu
+        
+        # pretrained_model is assumed to have checkpoint loaded
+        # if isinstance(pretrained_model, nn.Module):
+        #     pretrained_model.to(self.config.device)
+        #     self.pretrained_model = pretrained_model
+        # else:
+        #     raise ValueError("Pretrained Model is not subclass of torch.nn.Module")
+
+        # Model label
+        self.model_name = config.get('model', None)
+        if self.model_name is None:
+            self.model_name = self.pretrained_model.__class__.__name__
+
+        # We support two config modes
+        # (1) 'autoq' dict in NNCF compression.initializer
+        # (2) 'auto_quantization' as standalone dict in NNCF config
+        if 'autoq' == self.config.nncf_config.get('compression', {}).get('initializer', {}).get('precision', {}).get('type', {}):
+            self.autoq_cfg = self.config.nncf_config.get('compression', {}).get('initializer', {}).get('precision')
+        else:
+            self.autoq_cfg = self.config.nncf_config.get('auto_quantization')
+        self.finetune = self.autoq_cfg['finetune']
+
+        self.tie_quantizers = True # Default to associate same precision of quantizers that feed into the GEMM compute
+        if 'tie_quantizers' in self.autoq_cfg:
+            self.tie_quantizers = self.autoq_cfg['tie_quantizers']
+
+        self.skip_wall = False # Default to enable resource constraints
+        if 'skip_wall' in self.autoq_cfg:
+            self.tie_quantizers = self.autoq_cfg['skip_wall']
+
+        # Action space boundary - need to revise to work with discrete
+        self.action_bound = self._get_action_space(self.autoq_cfg)
+
+        # TODO (Design): How to generalize evaluation and train in general? tasks could have different metric and loss function
+        self._evaluate_pretrained_model()
+
+        # Quantizer Master Table Creation
+        qtracer = QuantizerTracer()
+        # qtable index is QID in string prepended with its quantize node id
+        qtable = qtracer.get_qtable(self.qctrl, self.qmodel)
+        qgroups = qtracer.get_quantizer_groups(qtable)
```

Comment:
We have a similar tested functionality (#191) in the develop branch for propagation mode: https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/precision_init/adjacent_quantizers.py#L26

Please consider reusing it

Reply:
High priority. Please discuss

Reply:
Refactored. QuantizerTracer is replaced with application of GroupsOfAdjacentQuantizers.